### DISCUSSION TITLE
How to set the keyword "ratio_failed"

### DISCUSSION THREAD

**Discussion by warriors1523** [deepmodeling/dpgen#763] [Q&A] [answered]
Hello everyone
     When I ran the model deviation at idx 0, I have the same problem as WenyeXuan (https://github.com/deepmodeling/dpgen/discussions/550). The problem is that some lammps calculations failed because of atoms lost. I wonder how to use keywords"ratio_failed" in the parameter file. 
     I use this keywords like this, but it doesn't seem to work. 
![image](https://user-images.githubusercontent.com/92622311/175773391-84fab70a-4a9a-452d-a8a3-841fe59ba534.png)

Looking forward to your reply.
Thanks very much !

**Comment by pee8379** [comment, marked as answer]
Looking for run.py, It seems ratio_failed only works for fp process.
If you use scheduler, with large number of jobs and small group_size, than you may try allow_failure option in machine json.

In my case, just manually delete failed model works fine.
When I met failed model_devi calculations, I delete those manually using simple python script after dpgen terminated, and restart dpgen again. 
Just make sure lmp failed due to temporary problem (poorly trained potential at first few iteration maybe?), not from user fault (strange structures, etc.).

**Reply by warriors1523** [reply]
Thank you for your answer!  Should I delete the failed tasks in "remote_root"(./model_work) or the tasks in iter.000000/01.model_devi or all of them when meet failed model_devi calculations? And should I change the serial number in the file(record.dpgen)?
![image](https://user-images.githubusercontent.com/92622311/175899316-0cb0e8e7-0739-4a53-a273-1952b82ba4c4.png)

**Comment by HuangJiameng** [comment]
1. You can attach (small) files to the section below or add URLs where to download an archive with all necessary files. Please try to create an input set that is as minimal and small as possible and reproduces the bug as quickly as possible. 2. You can change the serial number, and dpgen will restart at the step of the last number in 'record.dpgen'. 3. By the way, I am not sure if keyword 'ratio_failure' in machine.json can help you.

**Reply by HuangJiameng** [reply]
if you try to set 'ratio_failure', please checkout the devel branch https://github.com/deepmodeling/dpgen/pull/764, where the wrong spelling was fixed.
